src/Sylius/Bundle/GridBundle/DependencyInjection/Configuration.php

```diff
@@ -29,12 +29,41 @@ public function getConfigTreeBuilder()
         $treeBuilder = new TreeBuilder();
         $rootNode = $treeBuilder->root('sylius_grid');
 
+        $this->addEnabledDriversSection($rootNode);
         $this->addTemplatesSection($rootNode);
         $this->addGridsSection($rootNode);
 
         return $treeBuilder;
     }
 
+    private function addEnabledDriversSection(ArrayNodeDefinition $node)
+    {
+        // determine which drivers are distributed with this bundle
+        $driverDir = __DIR__ . '/../Resources/config/driver';
+        $iterator = new \RecursiveDirectoryIterator($driverDir);
+        foreach (new \RecursiveIteratorIterator($iterator) as $file) {
+            if ($file->getExtension() !== 'xml') {
+                continue;
+            }
+
+            $validDrivers[] = substr($file->getPathname(), 1 + strlen($driverDir), -4);
+        }
+
+        $node
+            ->children()
+                ->arrayNode('enabled_drivers')
```

Comment:
Well, I think the issue here is that we are just "turning on" drivers
which are distributed with the bundle. If the user has their own drivers
then they may think to add those here too, which would break things.

`drivers` implies all drivers, including those which are not distributed
with the bundle.

`enabled_drivers` is slightly more specific imo, not by much.

Ultimately I like the idea of the `driver provider` pattern better, in
which case all drivers would be turned on but lazily instantiated. But
this require more work.

But in general I don't really mind which term we use here.

On Mon, May 09, 2016 at 01:28:34AM -0700, Kamil Kokot wrote:
>    In [1]src/Sylius/Bundle/GridBundle/DependencyInjection/Configuration.php:
> 
>  > +    private function addEnabledDriversSection(ArrayNodeDefinition $node)
>  > +    {
>  > +        // determine which drivers are distributed with this bundle
>  > +        $driverDir = __DIR__ . '/../Resources/config/driver';
>  > +        $iterator = new \RecursiveDirectoryIterator($driverDir);
>  > +        foreach (new \RecursiveIteratorIterator($iterator) as $file) {
>  > +            if ($file->getExtension() !== 'xml') {
>  > +                continue;
>  > +            }
>  > +
>  > +            $validDrivers[] = substr($file->getPathname(), 1 + strlen($driverDir), -4);
>  > +        }
>  > +
>  > +        $node
>  > +            ->children()
>  > +                ->arrayNode('enabled_drivers')
> 
>    What do you think about renaming it to drivers instead?
> 
>    IMO this:
> 
>  sylius_grid:
>    drivers:
>      - doctrine/orm
> 
>    looks better than this:
> 
>  sylius_grid:
>    enabled_drivers:
>      - doctrine/orm
> 
>    And it is understandable easily as well
> 
>    —
>    You are receiving this because you were mentioned.
>    Reply to this email directly or [2]view it on GitHub
> 
>    Reverse link: [3]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/Sylius/Sylius/pull/4972#discussion_r62464191
>    2. https://github.com/Sylius/Sylius/pull/4972/files/63012337cc762730a69d5e45a0d5df9e0ef946fe#r62464191
>    3. https://github.com/Sylius/Sylius/pull/4972/files/63012337cc762730a69d5e45a0d5df9e0ef946fe#r62464191